docker/Dockerfile.py

```diff
@@ -24,15 +24,17 @@
 python = f'python{sys.version_info[0]}.{sys.version_info[1]}'
 pip = f'{python} -m pip'
 
-
-dependencies = ' '.join(['libffi-dev',  # For client side encryption for extras with PyNACL
+# Debian and Ubuntu don't package ensurepip by default so the python-venv package must be installed
+python_packages = {'python3.8': ['python3.8-distutils', 'python3.8-venv'],
+                   'python3.9': ['python3.9-distutils', 'python3.9-venv'],
+                   'python3.10': ['python3.10-distutils', 'python3.10-venv'],
+                   'python3.11': ['python3.11-distutils', 'python3.11-venv'],
+                   'python3.12': ['python3.12-distutils', 'python3.12-venv']}
+
+dependencies = ' '.join(python_packages[python] +
+                        ['libffi-dev',  # For client side encryption for extras with PyNACL
                          python,
                          f'{python}-dev',
-                         'python3.8-distutils' if python == 'python3.8' else '',
-                         'python3.9-distutils' if python == 'python3.9' else '',
-                         'python3.10-distutils' if python == 'python3.10' else '',
-                         'python3.11-distutils' if python == 'python3.11' else '',
```

Comment:
Should this just be `f'{python}-dev', f'{python}-distutils', f'{python}-venv'`? I think we used to have a python version that we needed that *didn't* have with the numbered packages available, but it seems like that's no longer the case.

Reply:
I separated the python packages into 2 as `python3.13-distutils` doesn't seem to exist in apt, and I'm not sure if it will be added as it was supposed to be [removed in Python 3.12](https://peps.python.org/pep-0632/).